tools/integration_tests/stale_handle/stale_file_handle_synced_file_test.go

```diff
@@ -0,0 +1,214 @@
+	assert.NoError(s.T(), err)
+
+	buffer := make([]byte, GCSFileSize)
+	err = operations.ReadBytesFromFile(fh, GCSFileSize, buffer)
+
+	operations.ValidateStaleNFSFileHandleError(s.T(), err)
+	ValidateObjectContentsFromGCS(ctx, storageClient, path.Join(testDirName, testCaseDir), FileName1, FileContents, s.T())
+}
+
+func (s *staleFileHandleSyncedFile) TestSyncedObjectClobberedRemotelyFirstWriteThrowsStaleFileHandleError() {
+	testCaseDir := "TestSyncedObjectClobberedRemotelyFirstWriteThrowsStaleFileHandleError"
+	targetDir := path.Join(testDirPath, testCaseDir)
+	operations.CreateDirectory(targetDir, s.T())
+	err := CreateObjectOnGCS(ctx, storageClient, path.Join(testDirName, testCaseDir, FileName1), GCSFileContent)
+	assert.NoError(s.T(), err)
+	filePath := path.Join(targetDir, FileName1)
+	fh, err := operations.OpenFileAsWriteOnly(filePath)
+	assert.NoError(s.T(), err)
+	// Replace the underlying object with a new generation.
+	err = WriteToObject(ctx, storageClient, path.Join(testDirName, testCaseDir, FileName1), FileContents, storage.Conditions{})
+	assert.NoError(s.T(), err)
+
+	_, err = fh.WriteString(Content)
+
+	operations.ValidateStaleNFSFileHandleError(s.T(), err)
+	// Attempt to sync to file should not result in error as we first check if the
+	// content has been dirtied before clobbered check in Sync flow.
+	operations.SyncFile(fh, s.T())
+	// Validate that object is not updated with new content as write failed.
+	ValidateObjectContentsFromGCS(ctx, storageClient, path.Join(testDirName, testCaseDir), FileName1, FileContents, s.T())
+}
+
+func (s *staleFileHandleSyncedFile) TestSyncedObjectClobberedRemotelySyncAndCloseThrowsStaleFileHandleError() {
+	testCaseDir := "TestSyncedObjectClobberedRemotelySyncAndCloseThrowsStaleFileHandleError"
+	targetDir := path.Join(testDirPath, testCaseDir)
+	operations.CreateDirectory(targetDir, s.T())
+	err := CreateObjectOnGCS(ctx, storageClient, path.Join(testDirName, testCaseDir, FileName1), GCSFileContent)
+	assert.NoError(s.T(), err)
+	filePath := path.Join(targetDir, FileName1)
+	fh, err := operations.OpenFileAsWriteOnly(filePath)
+	assert.NoError(s.T(), err)
+	// Dirty the file by giving it some contents.
+	_, err = fh.WriteString(Content)
+	assert.NoError(s.T(), err)
+	// Replace the underlying object with a new generation.
+	err = WriteToObject(ctx, storageClient, path.Join(testDirName, testCaseDir, FileName1), FileContents, storage.Conditions{})
+	assert.NoError(s.T(), err)
+
+	operations.SyncFileShouldThrowStaleHandleError(fh, s.T())
+	operations.CloseFileShouldThrowStaleHandleError(fh, s.T())
+
+	// Validate that object is not updated with un-synced content.
+	ValidateObjectContentsFromGCS(ctx, storageClient, path.Join(testDirName, testCaseDir), FileName1, FileContents, s.T())
+}
+
+func (s *staleFileHandleSyncedFile) TestSyncedObjectDeletedRemotelySyncAndCloseThrowsStaleFileHandleError() {
```

Comment:
This is same as clobbering the file by writing, I don't think we need a separate test case for this scenario. what do you think?

Reply:
Yeah, but we added this test to check that deleting locally as well as remotely has same effect.
Can remove it.